### DISCUSSION TITLE
Local IP on routers/gateways

### DISCUSSION THREAD

**Discussion by Guiorgy** [fastfetch-cli/fastfetch#1120] [General]
I just cross compiled fastfetch for OpenWRT, and the first thing that I thought was "why is Local IP showing my public IP?":

![openwrt](https://github.com/user-attachments/assets/87795435-0549-47e3-ab5e-b8cad821dfa5)

However, on second though, it makes sense, since to the gateway its "local" IP is the one the ISP gave it. But the local IP of the gateway from a (lan) clients perspective is its LAN IP:

![openwrt_ip](https://github.com/user-attachments/assets/85b8fe83-687a-4a7b-b1a5-1ca907389fab)

Just wanted to know what others thing about this.

Perhaps it would make sense to have an option/configuration to be able to tell fastfetch which interface is the "local" one?

**Comment by CarterLi** [comment]
By default, fastfetch shows the interface used for default route. Try this:

```jsonc
{
  "type": "localip",
  "defaultRouteOnly": false,
  "namePrefix": "br-lan"
}
```

**Reply by CarterLi** [reply]
It's not very common that ISP will give you a public IP address. Is your device used as for public servers?

**Reply by Guiorgy** [reply]
That did the trick. I should really give the [JSON Schema docs](https://github.com/fastfetch-cli/fastfetch/wiki/Json-Schema#modules_items_anyOf_i1_oneOf_i21_namePrefix) a thorough read at some point.

> It's not very common that ISP will give you a public IP address. Is your device used as for public servers?

Do you mean dynamic IPs under NAT?

**Reply by CarterLi** [reply]
Yes

**Reply by Guiorgy** [reply]
Nope, though it's a normal residential apartment, it's a fixed IP, which allows me to self host a Vaultwarden (Bitwarden) instance ¯\\\_(ツ)\_/¯

Maybe since the country is small they couldn't be bothered implementing a proper system, but that works for me, so whatever.

Edit: Found out that old residences that had their internet connected a long time ago have static IPs, but anything even remotely newer uses dynamic IPs. Looks like they can't be bothered touching and changing old equipment/configurations. "If it works, it works, so don't touch it" kind of situation :P

**Comment by PalebloodSky** [comment]
Nice work! Could you submit fastfetch to OpenWrt? Would love to have this in that project apk as it’s on my router. 

https://github.com/openwrt/packages
